core/vtk/ttkAlgorithm/ttkAlgorithm.cpp

```diff
@@ -382,6 +386,42 @@ int ttkAlgorithm::RequestDataObject(vtkInformation *ttkNotUsed(request),
   return 1;
 }
 
+#if TTK_ENABLE_MPI
+void ttkAlgorithm::MPIPreconditioning(vtkInformationVector **inputVector) {
```

Comment:
I'd change the prototype of this function to something more intuitive, such as:
```
void ttkAlgorithm::MPIPreconditioning(vtkDataSet **input) {
```

then, when calling this function, you'd need to do:
```
MPIPreconditioning(vtkDataSet::GetData(inputVector[0]));
```

I believe this is more intuitive. Plus, we really only need the dataset inside that function.